File: trunk/C#/RBD/RBD/NodeSyncInfo.cs
// + TODO check

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace RBD
{
    public class NodeSyncInfo
    {
        int vBeatsOutOfSync;
        public int BeatsOutOfSync
        {
            get
            {
                return vBeatsOutOfSync;
            }
        }

        //long LastBeat; // time in miliseconds
        DateTime LastBeat;

        public NodeSyncInfo()
        {
            InSync();
        }

        public void InSync()
        {
            vBeatsOutOfSync = 0;
            //LastBeat = System.currentTimeMillis();
            LastBeat = DateTime.Now;
        }

        public void BeatOutOfSync()
        {
            ++vBeatsOutOfSync;
            //LastBeat = System.currentTimeMillis();
            LastBeat = DateTime.Now;
        }

        public long getMsSinceLastBeat()
        {
            return (DateTime.Now - LastBeat).Milliseconds;
            //return System.currentTimeMillis() - LastBeat;
        }
    }
}

File: SR-RBD/src/ddb/NodeSyncInfo.java
package ddb;


public class NodeSyncInfo
{
	private int BeatsOutOfSync;
	private long LastBeat; // time in miliseconds
	
	public NodeSyncInfo()
	{
		InSync();
	}
	
	public void InSync()
	{
		BeatsOutOfSync = 0;
		LastBeat = System.currentTimeMillis();
	}
	
	public void BeatOutOfSync()
	{
		++BeatsOutOfSync;
		LastBeat = System.currentTimeMillis();
	}

	public int getBeatsOutOfSync() {
		return BeatsOutOfSync;
	}

	public long getMsSinceLastBeat() {
		return System.currentTimeMillis() - LastBeat;
	}
}